CONTRIBUTING.md

```diff
@@ -110,6 +110,29 @@ At this point, you should follow a standard git workflow:
 `make generate-in-docker` target which will install necessary tools in containerized environment and will generate assets.
 This targets needs `docker` to be installed on host and was not tested with other container runtime environments.
 
+## Testing
+
+Supposing $KUBECONFIOG is set to the config file of a Kubernetes cluster running the codes to test. We can run all tests by using this command `make test`.
+
+The testing consist of 3 aspects:
+- unit tests, can be run seperately by `make test-unit`.
+- Prometheus rule tests, can be run seperately by `make test-rules`.
+- end to end tests, can be run seperately by `make test-e2e`.
+Unit testing can be run by using `make test-unit` command.
+
+If we need to run a specific test case of the E2E test, we can use the following command.
+```bash
+go test -v -timeout=120m ./test/e2e/ --kubeconfig $KUBECONFIG -run TestBodySizeLimit
+```
+Attention that we have to pass a valid $KUBECONFIG explicitly `--kubeconfig $KUBECONFIG` even if ~/.kube/config exists.
+
+To run a specific test case of unit tests, we can refer to these commands.
+```bash
+go test -v -timeout=120m ./pkg/manifests -run TestImageParsing
+go test -v -timeout=120m ./pkg/client -run TestCreateOrUpdateService
+```
+Make sure change the root directory of the Golang package, the test function have to be named as `TestXXXXXX`.
```

Comment:
I'm not sure I fully understood  the first part of this phrase. Do you mean that to run the commands above we should be at the root of the golang package? In this case, inside the folder `cluster-monitoring-opetator`

Reply:
I should rephrase this line. I want to say to replace ./pkg/client to the root directory of the Go package (where we declare `package "XXX"` in the source code files).

Reply:
Just rephrased the line with an example. Does the new version offer a more clear explanation?

Reply:
Super clear now, thank you :)